test-lib.sh

```diff
@@ -1087,21 +1114,64 @@ ct_check_testcase_result() {
 # Uses: $TEST_SUMMARY - variable for storing test results
 # Uses: $IMAGE_NAME - name of the image being tested
 ct_run_tests_from_testset() {
-  local app_name="$1"
+  local app_name="${1:-appnamenotset}"
+  local time_beg_pretty
+  local time_beg
+  local time_end
+  local time_diff
+  local test_msg
+
+  # Let's store in the log what change do we test
+  echo
+  git show --no-patch
+  echo
+
   for test_case in $TEST_SET; do
     TESTCASE_RESULT=0
-    echo "Running test $test_case ... "
+    time_beg_pretty=$(ct_timestamp_pretty)
+    time_beg=$(ct_timestamp_s)
+    echo "-----------------------------------------------"
+    echo "Running test $test_case (starting at $time_beg_pretty) ... "
+    echo "-----------------------------------------------"
     $test_case
     ct_check_testcase_result $?
-    local test_msg
+    time_end=$(ct_timestamp_s)
     if [ $TESTCASE_RESULT -eq 0 ]; then
       test_msg="[PASSED]"
     else
       test_msg="[FAILED]"
       TESTSUITE_RESULT=1
     fi
-    printf -v TEST_SUMMARY "%s %s for '%s' %s\n" "${TEST_SUMMARY}" "${test_msg}" "${app_name}" "$test_case"
+    time_diff=$(ct_timestamp_diff "$time_beg" "$time_end")
+    printf -v TEST_SUMMARY "%s %s for '%s' %s (%s)\n" "${TEST_SUMMARY:-}" "${test_msg}" "${app_name}" "$test_case" "$time_diff"
+    [ -n "${FAIL_QUICKLY:-}" ] && return 1
   done;
 }
 
+# ct_timestamp_s
+# --------------
+# Returns timestamp in seconds
+function ct_timestamp_s() {
+  date '+%s'
+}
+
+# ct_timestamp_pretty
+# -----------------
+# Returns timestamp readable to a human
+function ct_timestamp_pretty() {
+  date --rfc-3339=seconds
+}
+
+# ct_timestamp_diff
+# -----------------
+# Computes a time diff between two timestamps
+# Argument: start_date - Beginning (in seconds)
+# Argument: final_date - End (in seconds)
+# Returns: Time difference in format HH:MM:SS
+function ct_timestamp_diff() {
+  local start_date=$1
+  local final_date=$2
+  date -u -d "0 $final_date seconds - $start_date seconds" +"%H:%M:%S"
```

Comment:
Can you please add to the comment an example of what is input, like format, and was is output?

Reply:
Output is described there: "Returns: Time difference in format HH:MM:SS"

Reply:
I've added some more info what exactly "seconds" means.